onnxruntime/core/framework/session_options.h

```diff
@@ -119,7 +120,13 @@ struct SessionOptions {
   std::unordered_map<std::string, const OrtValue*> initializers_to_share_map;
 
   // See onnxruntime_c_api.h for detailed documentation.
-  Status AddInitializer(_In_z_ const char* name, _In_ const OrtValue* val) noexcept;
+  Status AddInitializer(_In_z_ const char* name, _In_ const OrtValue* val);
+
+#if !defined(ORT_MINIMAL_BUILD)  && !defined(DISABLE_EXTERNAL_INITIALIZERS)
+  // Customer supplied pre-processed data for external initializers
+  InlinedHashMap<std::string, OrtValue> external_initializers;
```

Comment:
>OrtValue [](http://example.com/codeflow?start=30&length=8)

why OrtValue here and const OrtValue* in initializers_to_share_map?

Reply:
I prefer dealing with values rather than pointers. Copying OrtValue is cheap and it points to the same underlying buffer.